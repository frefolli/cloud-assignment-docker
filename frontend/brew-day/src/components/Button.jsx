import React, { useState } from "react";

const Button = () => {
    const [count, setCount] = useState(0);

    return (
        <React.Fragment>
            <span className = "shoppingItemText">{count}</span>
            <button onClick={() => {if (count > 0) setCount(count - 1)}}>-</button>
            <button onClick={() => setCount(count + 1)}>+</button>
        </React.Fragment>
    );
}

export default Button;
