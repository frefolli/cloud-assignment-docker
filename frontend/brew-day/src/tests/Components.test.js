import React from "react";
import { render } from '@testing-library/react';
import Button from '../components/Button';

describe('Button component', () => {
    test('should render correctly', () => {
      const { container } = render(<Button />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });