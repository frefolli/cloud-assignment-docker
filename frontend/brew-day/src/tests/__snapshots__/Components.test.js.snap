// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly 1`] = `
<span
  class="shoppingItemText"
>
  0
</span>
`;
